components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/api/builder/WSO2UserOperationEventPayloadBuilder.java

```diff
@@ -178,6 +178,67 @@ public EventPayload buildUserLockAccountEvent(EventData eventData) throws Identi
         return this.buildUserAccountEvent(eventData);
     }
 
+    @Override
+    public EventPayload buildUserProfileUpdateEvent(EventData eventData) throws IdentityEventException {
+
+        Map<String, Object> properties = eventData.getEventParams();
+        String tenantId = String.valueOf(properties.get(IdentityEventConstants.EventProperty.TENANT_ID));
+        String tenantDomain = String.valueOf(properties.get(IdentityEventConstants.EventProperty.TENANT_DOMAIN));
+        String userStoreDomainName =
+                String.valueOf(properties.get(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN));
+        String userId = String.valueOf(properties.get(IdentityEventConstants.EventProperty.USER_ID));
+
+        UserStore userStore = new UserStore(userStoreDomainName);
+
+        List<UserClaim> addedClaims = populateClaims(properties, IdentityEventConstants.EventProperty.USER_CLAIMS_ADDED);
+        List<UserClaim> modifiedClaims = populateClaims(properties, IdentityEventConstants.EventProperty.USER_CLAIMS_MODIFIED);
+        List<UserClaim> deletedClaims = populateClaims(properties, IdentityEventConstants.EventProperty.USER_CLAIMS_DELETED);
+        List<UserClaim> additionalClaims = populateClaims(properties, "ADDITIONAL_USER_CLAIMS");
+
+        User user = new User();
+        user.setId(userId);
+        user.setRef(
+                EventPayloadUtils.constructFullURLWithEndpoint(SCIM2_USERS_ENDPOINT) + "/" + user.getId());
+        user.setAdditionalClaims(additionalClaims);
+        user.setAddedClaims(addedClaims);
+        user.setUpdatedClaims(modifiedClaims);
+        user.setRemovedClaims(deletedClaims);
+
+        Organization organization = new Organization(tenantId, tenantDomain);
+
+        Flow flow = IdentityContext.getThreadLocalIdentityContext().getFlow();
+        String initiatorType = "";
+        String action = "";
+        if (flow != null) {
+            initiatorType = flow.getInitiatingPersona().name();
+            action = flow.getName().name();
```

Comment:
we can't get flow name as it is. I think we discussed this multiple times. Based on the flow we have to derive. 
e.g, if flow is user profile update -> action is 'UPDATE'
when it comes to user profile update with jit provisioning -> action is 'JIT'

Reply:
Fixed 75ee620374222089f9ec67652429e2be05a8ff19